src/Maker/MakeDecorator.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Bundle\MakerBundle\Util\DecoratorInfo;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
+
+/**
+ * @author Benjamin Georgeault <git@wedgesama.fr>
+ */
+final class MakeDecorator extends AbstractMaker
+{
+    /**
+     * @param array<string> $ids
+     */
+    public function __construct(
+        private readonly ContainerInterface $container,
+        private readonly array $ids,
+    ) {
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:decorator';
+    }
+
+    public static function getCommandDescription(): string
+    {
+        return 'Create CRUD for Doctrine entity class';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void
+    {
+        $command
+            ->addArgument('id', InputArgument::OPTIONAL, 'The ID of the service to decorate.')
+            ->addArgument('decorator-class', InputArgument::OPTIONAL, \sprintf('The class name of the service to create (e.g. <fg=yellow>%sDecorator</>)', Str::asClassName(Str::getRandomTerm())))
+            ->setHelp($this->getHelpFileContents('MakeDecorator.txt'))
+        ;
+    }
+
+    public function configureDependencies(DependencyBuilder $dependencies): void
+    {
+        $dependencies->addClassDependency(
+            AsDecorator::class,
+            'dependency-injection',
+        );
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        // Ask for service id.
+        if (null === $input->getArgument('id')) {
+            $argument = $command->getDefinition()->getArgument('id');
+
+            ($question = new Question($argument->getDescription()))
+                ->setAutocompleterValues($this->ids)
+                ->setValidator(fn ($answer) => Validator::serviceExists($answer, $this->ids))
```

Comment:
based on the issue I'd suggest that the command accept not only the id but a classname and tries to guess the id based on it (if the service does not exists for example)

Reply:
3 yes with fault tolerance maybe ? `serializer` vs `serialiser` There are example in symfony's code about this.

Reply:
Done. I also did guess from class/interface short name.